libtenzir/builtins/formats/xsv.cpp

```diff
@@ -179,10 +179,15 @@ struct xsv_printer_impl {
       TENZIR_UNREACHABLE();
     }
 
+    auto operator()(view<record>) noexcept -> bool {
+      TENZIR_UNREACHABLE();
+    }
```

Comment:
This revealed a bug in `flatten`, which did not work correctly for `{x: {}}`. This diff fixes it:

```diff
diff --git a/libtenzir/src/arrow_table_slice.cpp b/libtenzir/src/arrow_table_slice.cpp
index 700d2fa7f84..94c550f7be3 100644
--- a/libtenzir/src/arrow_table_slice.cpp
+++ b/libtenzir/src/arrow_table_slice.cpp
@@ -450,7 +450,7 @@ transform_columns(const table_slice& slice,
   if (slice.rows() == 0) {
     return {};
   }
-  if (caf::get<record_type>(slice.schema()).num_leaves() == 0) {
+  if (caf::get<record_type>(slice.schema()).num_fields() == 0) {
     return slice;
   }
   auto input_batch = to_record_batch(slice);
diff --git a/libtenzir/src/table_slice.cpp b/libtenzir/src/table_slice.cpp
index cef8bd80d3d..3d744b2f6ec 100644
--- a/libtenzir/src/table_slice.cpp
+++ b/libtenzir/src/table_slice.cpp
@@ -869,6 +869,9 @@ auto flatten_record(
   struct record_type::field field, const std::shared_ptr<arrow::Array>& array)
   -> indexed_transformation::result_type {
   const auto& rt = caf::get<record_type>(field.type);
+  if (rt.num_fields() == 0) {
+    return {};
+  }
   auto struct_array
     = std::static_pointer_cast<type_to_arrow_array_t<record_type>>(array);
   const auto next_name_prefix
@@ -965,7 +968,7 @@ auto make_rename_transformation(std::string new_name)
 } // namespace
 
 auto flatten(table_slice slice, std::string_view separator) -> flatten_result {
-  if (slice.rows() == 0 or slice.columns() == 0) {
+  if (slice.rows() == 0) {
     return {std::move(slice), {}};
   }
   // We cannot use arrow::StructArray::Flatten here because that does not
```

Reply:
Applied. Thanks for the patch!